azure/services/availabilitysets/availabilitysets.go

```diff
@@ -18,134 +18,98 @@ package availabilitysets
 
 import (
 	"context"
-	"strconv"
 
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-04-01/compute"
-	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/pkg/errors"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
-	"sigs.k8s.io/cluster-api-provider-azure/azure/converters"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/resourceskus"
+	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
+const serviceName = "availabilitysets"
+
 // AvailabilitySetScope defines the scope interface for a availability sets service.
 type AvailabilitySetScope interface {
 	azure.ClusterDescriber
-	AvailabilitySet() (string, bool)
+	azure.AsyncStatusUpdater
+	AvailabilitySetSpec() azure.ResourceSpecGetter
 }
 
 // Service provides operations on Azure resources.
 type Service struct {
 	Scope AvailabilitySetScope
 	Client
+	async.Reconciler
 	resourceSKUCache *resourceskus.Cache
 }
 
 // New creates a new availability sets service.
 func New(scope AvailabilitySetScope, skuCache *resourceskus.Cache) *Service {
+	Client := NewClient(scope)
```

Comment:
```suggestion
	client := NewClient(scope)
```

this var doesn't need to be exported